src/temporal_logic/rules.rs

```diff
@@ -1179,6 +1194,26 @@ macro_rules! invariant_action_n_internal {
 pub use invariant_action_n;
 pub use invariant_action_n_internal;
 
+#[macro_export]
```

Comment:
Could you document this one?

Reply:
Temporarily not. Later I want to combine `strengthen_next`, `invariant_action` and `invariant_next` more compact. They all need to show:

```
spec |= []p1,
spec |= []p2
...
spec |= []pn

post:
spec |= [](p1 /\ p2 /\ ... /\ pn)
```

Reply:
This is a cool lemma but I find the description confusing. Besides the `==` thing mentioned above (apologize that I am one who introduced `==` into the description of `strengthen_next_n`), can you:
- describe some common use cases of this lemma, like the ones in invariants.rs
- explain that `partial_spec |= inv` is often proved by sth like `forall |s, s_prime| partial_spec(s, s_prime) ==> inv(s, s_prime)`
- explain the difference between this one and `init_invariant` -- people might get confused by these two tho they should be used in very difference cases
- see if you can use a different term than `partial_spec` -- this is a very confusing term (sorry again I am the one who introduced the term). You can see if you can find some better terms; I know `stronger_next` is not precise but I am OK with it.